facia/app/utils/FaciaPicker.scala

```diff
@@ -8,24 +8,39 @@ import play.api.mvc.RequestHeader
 
 class FaciaPicker extends GuLogging {
 
-  // we don't yet use these parameters but they will play into dcrCouldRender later on
+  val SUPPORTED_COLLECTIONS: Set[String] =
+    Set("dynamic/fast", "dynamic/slow", "fixed/small/slow-IV", "fixed/large/slow-XIV")
+
+  def dcrSupportsAllCollectionTypes(faciaPage: PressedPage): Boolean = {
+    faciaPage.collections.forall(collection => SUPPORTED_COLLECTIONS.contains(collection.collectionType))
+  }
```

Comment:
So does this overall function return true is ALL the items in `faciaPage.collections` return true?

Um, so in Javascript you can use [every](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every) for this kind of thing.

Is this... can it be... surely not... but is this a case where javascript is... and I still can't believe it.... better than Scala?

Reply:
Don’t fret @oliverlloyd, Scala is mighty:

```scala
def dcrSupportsAllCollectionTypes(faciaPage: PressedPage): Boolean = {
  faciaPage.collections.toSet.subsetOf(SUPPORTED_COLLECTIONS)
}
```

Reply:
🤯

Reply:
```ts
function supportsAll(collections) {
  collections.every(collection => SUPPORTED_COLLECTIONS.includes(collection.name));
}
```

Looks nicer to me

![who-knows-shrug](https://user-images.githubusercontent.com/1336821/168116133-20df96a3-6217-4f0c-ad45-b784cfe19f24.gif)

Reply:
I love `every`, but your comparison is unfair. If you'd have to turn a faciaPress into a `Set` in JS you would have the same amount of verbosity.

Also worth noting that if you want to keep things pure, you should pass in the set of supported elements as an argument.

```scala
def supportsAll(collection, supported) {
  collection.subsetOf(supported)
}

```